typescript/sdk/src/router/schemas.ts

```diff
@@ -17,8 +17,17 @@ export const ForeignDeploymentConfigSchema = z.object({
   foreignDeployment: z.string().optional(),
 });
 
+export const RemoteRouterSchema = z.record(
+  z.string(), // domain
+  z.string(), // router
```

Comment:
Why not name them here?

Reply:
What did you have in mind to name it?

I want it match the contract's `mapping(uint domain => bytes32 router)`

In this case, the structure for this is object is
```
{ 
    domain1: 'some router address as bytes'),
    domain2: 'some router address as bytes')
    domain3: 'some router address as bytes')
}
```

Reply:
something simple like
```suggestion
  const RemoteRouterDomain = z.string();
  const RemoteRouterRouter = z.string().startsWith('0x');

  export const RemoteRouterSchema = z.record(
    RemoteRouterDomain,
    RemoteRouterRouter,
  );
```

Reply:
Partially updated. going to remove the explicit "Domain" schema type as its not really needed. Also removed the comments

Reply:
after reading the discord comments. i think its fair to say that the suggested code is clearer. 

will do that instead :)